app/services/foreman_ansible/playbooks_importer.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  # Imports playbooks from smart proxy
+  class PlaybooksImporter
+    include ::ForemanAnsible::ProxyAPI
+    delegate :playbooks, :playbooks_names, :to => :proxy_api
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def import_playbooks(playbooks_names)
+      filter_playbooks = playbooks_names.empty? ? playbooks : PlaybooksImporterHelper.find_playbooks(playbooks, playbooks_names)
+      import_playbooks_content filter_playbooks
+    end
+
+    def import_playbooks_content(playbooks)
+      params = { changed: { update: [], create: [] } }
+      playbooks.each do |playbook|
+        new_playbook = parse_playbook playbook
+        job_template_id = JobTemplate.where(name: playbook['name']).pluck(:id).first
+        old_var_block = PlaybooksImporterHelper.check_vars_block new_playbook[:old_vars_block]
+        if job_template_id.present?
+          params[:changed][:update].push(
+            {
+              job_template_id: job_template_id,
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        else
+          params[:changed][:create].push(
+            {
+              playbook: new_playbook,
+              old_var_block: old_var_block
+            }
+          )
+        end
+      end
+      params
+    end
+
+    def parse_playbook(playbook)
+      playbook_name =  playbook['name']
+      metadata = add_metadata playbook['name']
+      new_playbook_content = convert_to_job_template_input playbook['playbooks_content']
+      {
+        name: playbook_name,
+        playbook_content: metadata + new_playbook_content[:playbook_content],
+        template_inputs: new_playbook_content[:template_inputs],
+        old_vars_block: new_playbook_content[:old_vars_block]
+      }
+    end
+
+    def add_metadata(playbook_name)
+      <<-METADATA
+      <%#
+        name: #{playbook_name}
+        snippet: false
+        job_category: Ansible Playbook - Imported
+        provider_type: Ansible
+        kind: job_template
+        model: JobTemplate
+      %>
+      METADATA
```

Comment:
heredoc maintains the whitespace, so keeping the indentation will add 6 spaces before `<%#`

```ruby
    def add_metadata(playbook_name)
<<-METADATA
<%#
  name: #{playbook_name}
  snippet: false
  job_category: Ansible Playbook - Imported
  provider_type: Ansible
  kind: job_template
  model: JobTemplate
%>
METADATA
    end
```

If you think this has poor readability, then we probably need to control the whitespace in s different way:

```ruby
def add_metadata(playbook_name)
      [
        '<%#',
        "  name: #{playbook_name}",
        "  snippet: false",
        "  job_category: Ansible Playbook - Imported",
        "  provider_type: Ansible",
        "  kind: job_template",
        "  model: JobTemplate",
        "%>"
      ].join("\n")
    end
```